test/functional/web/safari-nativewebtap-e2e-specs.js

```diff
@@ -1,270 +1,258 @@
+import {
+  openPage, spinTitleEquals, spinTitle, GUINEA_PIG_PAGE,
+  GUINEA_PIG_SCROLLABLE_PAGE, GUINEA_PIG_APP_BANNER_PAGE
+} from './helpers';
+import { retryInterval } from 'asyncbox';
+import B from 'bluebird';
+import {CLASS_CHAIN_SEARCH} from '../helpers/element';
+
+chai.should();
+chai.use(chaiAsPromised);
+
+const caps = amendCapabilities(SAFARI_CAPS, {
+  'appium:safariInitialUrl': GUINEA_PIG_PAGE,
+  'appium:nativeWebTap': true,
+  'appium:usePrebuiltWDA': hasDefaultPrebuiltWDA(),
+});
+
+const SPIN_RETRIES = 25;
+
+const PAGE_3_LINK = 'i am a link to page 3';
+const PAGE_3_TITLE = 'Another Page: page 3';
+
+describe('Safari - coordinate conversion -', function () {
+  this.timeout(MOCHA_TIMEOUT * 2);
+
+  let devices = [];
+  before(function () {
+    if (process.env.REAL_DEVICE) {
+      // skip, by not having any devices in the list
+    } else {
+      devices = [DEVICE_NAME, DEVICE_NAME_FOR_SAFARI_IPAD];
+    }
+
+    async function loadPage (driver, url) {
+      await retryInterval(5, 1000, async function () {
+        await openPage(driver, url);
+        await spinTitle(driver).should.eventually.not.include('Cannot Open Page');
+      });
+    }
+
+    // Close all tabs in Safari before running tests because left over tabs can affect the test results.
+    // If possible, it's probably better to kick com.apple.mobilesafari.settings.DeleteAllDataAndCachesTask
+    // task in com.apple.Preferences app or something equivalent rather than closing tabs via GUI.
+    async function closeAllTabsViaSettingsApp (deviceName) {
+      const newCaps = {
+        'appium:deviceName': deviceName,
+        'appium:usePrebuiltWDA': hasDefaultPrebuiltWDA(),
+      };
+      const localSettingsCaps = amendCapabilities(SETTINGS_CAPS, newCaps);
+      const driver = await initSession(localSettingsCaps);
+      await driver.$(CLASS_CHAIN_SEARCH + ':**/XCUIElementTypeStaticText[`label == "Safari"`]').click();
+      await driver.$('~CLEAR_HISTORY_AND_DATA').click();
+      if (await driver.$$('~Clear').length > 0) { // for iPad
+        await driver.$('~Clear').click();
+      } else { // for iPhone
+        await driver.$('~Clear History and Data').click();
+      }
+      if (util.compareVersions(extractCapabilityValue(localSettingsCaps, 'appium:platformVersion'), '>=', '16.0')) {
+        await driver.$(CLASS_CHAIN_SEARCH + ':**/XCUIElementTypeButton[`label == "Close Tabs"`]').click();
+      }
+      await deleteSession();
+    }
+
+    // define the tests, for each device
+    for (const deviceName of devices) {
+      describe(`${deviceName} -`, function () {
+        this.timeout(MOCHA_TIMEOUT * 2);
+
+        let driver;
+        const localCaps = amendCapabilities(caps, {'appium:deviceName': deviceName});
+        let skipped = false;
+
+        before(async function () {
+          skipped = false;
+          try {
+            await closeAllTabsViaSettingsApp(deviceName);
```

Comment:
I changed my mind and modified to close the safari tabs via GUI. 

[MEMO]
1. As for https://appium.github.io/appium-xcuitest-driver/4.16/execute-methods/#mobile-updatesafaripreferences, I tried removing the contents of `BrowserControllersSavedState` property in `Library/Preferences/com.apple.mobilesafari.plist` file but it doesn't help. 
2. While playing with Preferences app, I saw a line of device log like `default	22:07:24.666146+0900	Preferences	-[UIApplication _beginBackgroundTaskWithName:expirationHandler:]: Will add backgroundTask with taskName: com.apple.mobilesafari.settings.DeleteAllDataAndCachesTask, expirationHandler: <__NSMallocBlock__: 0x600001680a20>`. I guess it is probably a function embedded in `com.apple.Preferences` app which closes safari tabs. If it's possible to kick via CUI, it's much better but I don't have any good idea.